core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultExternalConfigSupplier.java

```diff
@@ -81,16 +86,21 @@ public VaultExternalConfigSupplier(ManagementContext managementContext, String n
             this.version = -1; // satisfy the static analysis :)
             errors.add("'kv-api-version' must be either 1 or 2");
         }
+        recoverTryCount = NumberUtils.toInt(config.get("recoverTryCount"), 10);
         mountPoint = config.get("mountPoint");
         if (Strings.isBlank(mountPoint) && this.version == 2) errors.add("missing configuration 'mountPoint'");
-        if (!Strings.isBlank(mountPoint) && this.version == 1) errors.add("'mountPoint' is only applicable when kv-api-version=2");
+        if (!Strings.isBlank(mountPoint) && this.version == 1)
```

Comment:
there is some ugly old code here, using `this.version` as an `int` but `String version` local var shadowing it.  not your fault but if you encounter it and hate it as much as me feel free to improve in future!